vnext/Microsoft.ReactNative.Cxx/JsiAbiApi.h

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+#ifndef MICROSOFT_REACTNATIVE_JSIABIAPI
+#define MICROSOFT_REACTNATIVE_JSIABIAPI
+
+#include <mutex>
+#include <unordered_map>
+#include "Crash.h"
+#include "jsi/jsi.h"
+#include "winrt/Microsoft.ReactNative.h"
+
+namespace winrt::Microsoft::ReactNative {
+
+// Forward declare the facebook::jsi::Runtime implementation on top of ABI-safe IJsiRuntime.
+struct JsiAbiRuntime;
+
+// Used as an ABI-safe opaque value for pointer values.
+using JsiPointerHandle = uint64_t;
+
+// An ABI-safe wrapper for facebook::jsi::Buffer.
+struct JsiBufferWrapper : implements<JsiBufferWrapper, IJsiBuffer> {
+  JsiBufferWrapper(std::shared_ptr<facebook::jsi::Buffer const> const &buffer) noexcept;
+  ~JsiBufferWrapper() noexcept;
+  uint32_t Size();
+  void GetData(JsiDataHandler const &handler);
+
+ private:
+  std::shared_ptr<facebook::jsi::Buffer const> m_buffer;
+};
+
+// A wrapper for ABI-safe JsiPreparedJavaScript.
+struct JsiPreparedJavaScriptWrapper : facebook::jsi::PreparedJavaScript {
```

Comment:
>facebook::jsi::PreparedJavaScript [](start = 38, length = 33)

Could you perhaps add a comment explaining the relationship between the constructs here and the  Microsoft.ReactNative.*Jsi* components in JsiApi.idl? I would have expected the constructs here to be build on top of Microsoft.ReactNative.*Jsi* components in JsiApi.idl, not on top of facebook::jsi components.

Reply:
Yes, I will add it. Whether we implement JSI or JsiAPI.idl depends on the operation. When we implement the facebook::jsi::Runtime::prepareJavaScript we expect to return a class that inherits from facebook::jsi::PreparedJavaScript, but ABI uses JsiPreparedJavaScript, thus we need to wrap up the JsiPreparedJavaScript into the facebook::jsi::PreparedJavaScript implementation to return it.

---
In reply to: [451866330](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r451866330) [](ancestors = 451866330)